src/attack.c

```diff
@@ -591,15 +593,24 @@ check_host_authorization (gvm_host_t *host, const struct in6_addr *addr,
   hosts_deny = gvm_hosts_new (prefs_get ("hosts_deny"));
   if (!host_authorized (host, addr, hosts_allow, hosts_deny))
     {
-      error_message_to_client2 (kb, "Host access denied.", NULL);
+      char ip_str[INET6_ADDRSTRLEN];
+
+      addr6_to_str (addr, ip_str);
+      error_message_to_client2 (kb, "Host access denied.", ip_str, NULL);
+      kb_item_set_str (kb, "internal/host_deny", "True", 0);
       return -1;
     }
   sys_hosts_allow = gvm_hosts_new (prefs_get ("sys_hosts_allow"));
   sys_hosts_deny = gvm_hosts_new (prefs_get ("sys_hosts_deny"));
   if (!host_authorized (host, addr, sys_hosts_allow, sys_hosts_deny))
     {
+      char ip_str[INET6_ADDRSTRLEN];
+
+      addr6_to_str (addr, ip_str);
       error_message_to_client2 (
-        kb, "Host access denied (system-wide restriction.)", NULL);
+        kb, "Host access denied (system-wide restriction.)", ip_str, NULL);
+      kb_item_set_str (kb, "internal/host_deny", "True", 0);
```

Comment:
This is done for every host (in `struct host *hosts`). I think it would be better to include this in the host directly and not via Redis. Maybe something like `host->host_deny` or better `host->access_allowed` (so its harder to missunderstand).

E.g. in `hosts_new()` do `h->access_allowed = FALSE;` and then in `attack_start()` do something like
```
  if (check_host_authorization (args->host, &hostip, kb))
    {
      error_message_to_client2...
      return;
    }
  else 
      hosts_set_access_allowed(name);
```

Reply:
This is not possible, because the check is done from the host process and the modified struct will not be change the data in the accessed by host_read(), which is called for the parent process. Therefore, the communication is done through redis.